### ISSUE TITLE
Clean up crate feature naming

### ISSUE THREAD

**Issue by ctz** [rustls/rustls#1961]
- We should remove the `logging` feature, and instead use the `log` implicit feature.
- We should remove the `aws_lc_rs` feature, and instead use the `aws-lc-rs` implicit feature.

**Comment by stormshield-gt** [comment]
I think it worth mentioning that [2024 edition will remove implicit features](https://doc.rust-lang.org/nightly/edition-guide/rust-2024/cargo-remove-implicit-features.html)

**Comment by djc** [comment]
Yes, although I think we should use explicit features bearing the exact name of the dependency.
